projects/notebooks/visualizations/rhods-notebooks-performance/analyze/__init__.py

```diff
@@ -1,13 +1,39 @@
-import matrix_benchmarking.common as common
+import json
 import logging
+import numpy as np
+
+from copy import deepcopy
+
+import matrix_benchmarking.common as common
+import matrix_benchmarking.regression as regression
 
 def run():
-    logging.info(f"Received {common.Matrix.count_records()} new entries")
-    for entry in common.Matrix.all_records(): pass
 
     logging.info(f"Received {common.LTS_Matrix.count_records()} historic LTS entries")
-    for lts_entry in common.LTS_Matrix.all_records(): pass
+    lts_entries_raw = list(common.LTS_Matrix.all_records())
+    # Temporarily skip the gathered results
+    lts_entries = list(filter(lambda entry: type(entry.results) is not list, lts_entries_raw))
+
+    logging.info(f"Received {common.Matrix.count_records()} new entries")
+    new_entries = list(common.Matrix.all_records())
 
     number_of_failures = 0
+    for entry in common.Matrix.all_records():
+        regression_results_dest = entry.location / "regression.json"
+
+        # Check for regression over the image
+        zscore = regression.ZScoreIndicator(
+            entry,
+            lts_entries,
+            settings_filter={"image": "tensorflow:2023.1"},
```

Comment:
can you try to make this dynamic?
`common.Matrix.settings["image"]` should contain all the available images

Reply:
could you try to do the filtering outside of the `regression.ZScoreIndicator` function ? 🤔 
I think it would be clearer if you could pass only the relevant `lts_entries`

Reply:
Sure, I'll add the filtering outside first. It should be as simple as removing the `settings_filter` option. How do you mean dynamic though? As in, we want to run a regression analysis for each unique "image"?